packages/mui-base/src/MultiSelectUnstyled/MultiSelectUnstyled.types.ts

```diff
@@ -106,37 +106,34 @@ export interface MultiSelectUnstyledType {
       component: C;
     } & OverrideProps<MultiSelectUnstyledTypeMap<TValue>, C>,
   ): JSX.Element | null;
-  <TValue extends {}>(
-    props: DefaultComponentProps<MultiSelectUnstyledTypeMap<TValue>>,
-  ): JSX.Element | null;
+  <TValue>(props: DefaultComponentProps<MultiSelectUnstyledTypeMap<TValue>>): JSX.Element | null;
   propTypes?: any;
 }
 
-export interface MultiSelectUnstyledOwnerState<TValue extends {}>
-  extends MultiSelectUnstyledProps<TValue> {
+export interface MultiSelectUnstyledOwnerState<TValue> extends MultiSelectUnstyledProps<TValue> {
```

Comment:
For all those changes, I don't understand the typing.
We have `TValue extends {}`.
But in the example we often pass strings.
Is this constrain wrong ?
Should we put another constraint ?

With the prop types correctly generated, I had test failing.

Reply:
`string` does extend `{}`. `T extends {}` basically means you can't provide `null` or `undefined` to the generic parameter.

Reply:
I think a better solution would be to detect the `extends {}` constraint and generate the PropTypes.any constraint from it.

Reply:
Done :+1: